agent/consul/state/peering.go

```diff
@@ -251,45 +251,42 @@ func (s *Store) peeringSecretsWriteTxn(tx WriteTxn, secret *pbpeering.PeeringSec
 
 	var toDelete []string
 	if existing != nil {
```

Comment:
Should we add validation that existing != nil unless the operation is generatetoken?

Reply:
Yep put that in